src/server/game/Spells/Spell.h

```diff
@@ -274,13 +274,21 @@ class TC_GAME_API SpellCastTargets
         std::string m_strTarget;
 };
 
+enum SpellValueCritFlag : uint8
```

Comment:
It is a valid way of making sure that the value returned by `CritFlag` via `SpellValueCritFlag` is `uint8` instead of regular `int`. Check how they are used in `Spell.cpp` above and notice how `CritFlag` and `SpellValueCritFlag` don't need to be defined as `uint8` there.